src/vs/workbench/contrib/chat/browser/chatWidget.ts

```diff
@@ -2188,6 +2192,16 @@ export class ChatWidget extends Disposable implements IChatWidget {
 		this.container.setAttribute('data-session-id', model.sessionId);
 		this.viewModel = this.instantiationService.createInstance(ChatViewModel, model, this._codeBlockModelCollection);
 
+		// Contributed session model handling: register contributed models with unified language model service
+		// TODO: This needs to be 'reset' or if will affect new chat editors. Understand this more.
+		const sessionCtx = this.chatService.getChatSessionFromInternalId(model.sessionId);
+		if (sessionCtx && sessionCtx.chatSessionType !== 'local') {
+			const contributedModels = this.chatSessionsService.getModelsForSessionType(sessionCtx.chatSessionType) ?? [];
+			this.languageModelsService.setContributedSessionModels(sessionCtx.chatSessionType, contributedModels);
```

Comment:
Ideally the view wouldn't have to pass these models around, can the ChatService or something else be responsible for syncing these into the LM service?

Reply:
I assume what we want here is similar to what mode does which is one place where the picker gets filtered down to the capable models. Unless you're proposing that the language model service starts accepting mode + session and filters based on those params?